server/src/test/java/org/elasticsearch/rest/RestControllerTests.java

```diff
@@ -640,6 +641,41 @@ public boolean compatibilityRequired() {
         assertTrue(channel.getSendResponseCalled());
     }
 
+    public void testIncorrectCompatibleHandlersDoNotDispatch() {
+        final byte version = (byte) (Version.CURRENT.major - 1);
+
+        final String mimeType = randomFrom("application/vnd.elasticsearch+json;compatible-with="+version);
```

Comment:
don't think you need the randomFrom

Reply:
refactored this to actually use a random subtype (yml, cbor, etc)